### PR TITLE
Add C# code to array doc

### PR DESCRIPTION
**Pull request by mingXta**
Add the C# code to docs(chapter of array)

### CODE REVIEW

**Review by krahets — COMMENTED** [review]
Thanks @mingXta! Could you make a source code file in `codes/csharp/chapter_array_and_linkedlist/array.cs` and test it?

Besides docs, This book provides executable source codes as well.

**Comment by mingXta** [comment]
> Thanks @mingXta! Could you make a source code file in `codes/csharp/chapter_array_and_linkedlist/array.cs` and test it?
> 
> Besides (除了) docs, This book provides (提供) executable source codes as well.

I've added it

**Comment by mingXta** [comment]
@krahets  I've added it,and I'll add the rest of the C# code and doc.

File: docs/chapter_array_and_linkedlist/array.md [hunk 1]
```diff
@@ -505,7 +544,21 @@ elementAddr = firtstElementAddr + elementLength * elementIndex
 === "C#"
 
     ```csharp title="array.cs"
-    
+     /* 遍历数组 */
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:547]: Incorrect indentation

File: codes/c#/chapter_array_and_linkedlist/Array.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
+﻿namespace hello_algo.chapter_arrag_and_linkedlist
```
Comment (krahets) [inline codes/c#/chapter_array_and_linkedlist/Array.cs:1]: Please add a file header following the java code.

File: codes/c#/chapter_array_and_linkedlist/Array.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+{
+    public class Array
+    {
+        /* 随机返回一个数组元素 */
+
```
Comment (krahets) [inline codes/c#/chapter_array_and_linkedlist/Array.cs:6]: Remove the empty line.

File: docs/chapter_array_and_linkedlist/array.md [hunk 4]
```diff
@@ -169,7 +173,14 @@ elementAddr = firtstElementAddr + elementLength * elementIndex
 === "C#"
 
     ```csharp title="array.cs"
-    
+     /* 随机返回一个数组元素 */
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:176]: Incorrect indentation.

**Pull request merged**